jerry-ext/debugger/debugger-tcp.c

```diff
@@ -150,64 +226,105 @@ jerryx_debugger_tcp_receive (jerry_debugger_transport_header_t *header_p, /**< t
 } /* jerryx_debugger_tcp_receive */
 
 /**
- * Create a tcp connection.
+ * Utility method to prepare the server socket to accept connections.
  *
- * @return true if successful,
- *         false otherwise
+ * The following steps are performed:
+ *  * Configure address re-use.
+ *  * Bind the socket to the given port
+ *  * Start listening on the socket.
+ *
+ * @return true if everything is ok
+ *         false if there was an error
  */
-bool
-jerryx_debugger_tcp_create (uint16_t port) /**< listening port */
+static bool
+jerryx_debugger_tcp_configure_socket (jerryx_socket server_socket, /** < socket to configure */
+                                      uint16_t port) /** < port number to be used for the socket */
 {
-  int server_socket;
   struct sockaddr_in addr;
-  socklen_t sin_size = sizeof (struct sockaddr_in);
 
   addr.sin_family = AF_INET;
   addr.sin_port = htons (port);
   addr.sin_addr.s_addr = INADDR_ANY;
 
-  if ((server_socket = socket (AF_INET, SOCK_STREAM, 0)) == -1)
+  int opt_value = 1;
+
+  if (setsockopt (server_socket, SOL_SOCKET, SO_REUSEADDR, &opt_value, sizeof (int)) != 0)
   {
-    JERRYX_ERROR_MSG ("Error: %s\n", strerror (errno));
     return false;
   }
 
-  int opt_value = 1;
+  if (bind (server_socket, (struct sockaddr *)&addr, sizeof (struct sockaddr)) != 0)
```

Comment:
According to the docs of `bind` the 3rd arg should be the size of the 2nd arg. But `addr` is of type `struct sockaddr_in`, not `struct sockaddr`.

Reply:
Ahh yeah. Currently it works as the `sockaddr_in` can be considered as a "subclass" of the `sockaddr` and they have the same size (at least on linux). But will change it to `sizeof (struct sockaddr_in)`